internal/engine/resourceset.go

```diff
@@ -39,34 +40,36 @@ type ResourceSet struct {
 // GetResources converts the CUE value to a list of ResourceSets.
 func GetResources(value cue.Value) ([]ResourceSet, error) {
 	var sets []ResourceSet
-	iter, _ := value.Fields(cue.Concrete(true))
+	iter, err := value.Fields(cue.Concrete(true), cue.Final())
+	if err != nil {
+		return nil, fmt.Errorf("getting resources failed, error: %w", err)
+	}
 	for iter.Next() {
 		name := iter.Selector().String()
 		expr := iter.Value()
-		switch expr.Kind() {
-		case cue.ListKind:
-			items, err := expr.List()
-			if err != nil {
-				return nil, fmt.Errorf("listing objects for %s failed, error: %w", name, err)
-			}
+		if expr.Err() != nil {
+			return nil, fmt.Errorf("getting value of resource list %q failed, error: %w", name, expr.Err())
```

Comment:
Not related to this PR, but something that I noticed while going over this. What's the benefit of suffixing with `, error:` when doing error wrapping? Given it does not make much sense for the individual error without the context from the stack of wrapped errors.

cc: @stefanprodan

Reply:
It's just my bad habit..  should be `failed: %w`?

Reply:
Yes, because then if you "detach" the error from the stack it looks like a thing on it's own without giving you the idea something is missing :smile: 

> getting value of resource list %q failed